### PR TITLE
Tools/Extractor: mapextractor & vmap4extractor now detects locales

### PR DESCRIPTION
**Pull request by robertrpf**
**Changes proposed:**

-  Add a GetLocale function that get the locale index and solves the Unable to open errors.
-  

**Target branch(es):** 3.3.5/master

- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)

- No open issues as far as I know.

**Tests performed:** Extracted maps and vmaps twice in a esEs locale.

**Known issues and TODO list:** (add/remove lines as needed)
- [x] Codestyle cleanup (convert tabs to spaces, remove blank lines)

### CODE REVIEW

File: src/tools/map_extractor/System.cpp [hunk 1]
```diff
@@ -1292,6 +1292,31 @@ void ExtractGameTables()
     printf("Extracted %u files\n\n", count);
 }
 
+
+int GetLocale() {
+
+	int i = 0;
+	int result = -1;
+
+	if (!((CONF_Locale && !(CONF_Locale & (1 << i))) && (i == LOCALE_none))){
```
Comment (jackpoz) [inline src/tools/map_extractor/System.cpp]: local variable i is assigned to a constant value so "1 << i" can be replaced with "1" and "i == LOCALE_none" can be removed (assuming LOCALE_none is defined as = 0)

**Comment by jackpoz** [comment]
what kind of tests were executed related to this PR ?

File: src/tools/vmap4_extractor/vmapexport.cpp [hunk 2]
```diff
@@ -141,8 +141,36 @@ uint32 WowLocaleToCascLocaleFlags[12] =
     CASC_LOCALE_ITIT,
 };
 
+
+int GetLocale() {
+
+	int i = 0;
+	int result = -1;
+
+	if (!(i == LOCALE_none)) {
```
Comment (jackpoz) [inline src/tools/vmap4_extractor/vmapexport.cpp]: "i" value is always 0

**Comment by jackpoz** [comment]
not sure if the "if" checks before the "while" loops are supposed to be moved somewhere else, they seem to be constant

**Comment by Robertrpf** [comment]
Thank you @jackpoz I didn't realized about conditionals, i only changed the code i needed in the first commit.

I changed too the file i was checking in vmaps.

For me is working with esES locale, I tried all the process twice.

**Comment by Carbenium** [comment]
And I see a lot of tabsss

**Comment by tkrokli** [comment]
Does this PR solve (close) or update issue **[mapextractor is not extracting specific .db2 files correctly](https://github.com/TrinityCore/TrinityCore/issues/19889)** **#19889** ? If yes, please add that info in your PR description, section "**Issues addressed:** Closes # (insert issue tracker number)".

It would also be nice to have a bit more details in the PR description, a bit more than just the PR title.

**Comment by Robertrpf** [comment]
@Carbenium for the tabs i followed the [guide](https://trinitycore.atlassian.net/wiki/pages/viewpage.action?pageId=2130103) , if it's needed i can recheck the options and fix it.

@tkrokli I don't know if it fixs the issue  [mapextractor is not extracting specific .db2 files correctly #19889](https://github.com/TrinityCore/TrinityCore/issues/19889), i couldn't reproduce it.

**Comment by tkrokli** [comment]
Yes, you need to fix your local options, so tabs are always converted to 4 spaces.
All your indents are currently tabs, which GitHub shows us as 8 spaces in width.
Also, please remove redundant blank lines (several of them contain whitespace).

File: src/tools/vmap4_extractor/vmapexport.cpp [hunk 3]
```diff
@@ -141,6 +141,29 @@ uint32 WowLocaleToCascLocaleFlags[12] =
...
+        {
+            result = i;
+        }
+            i++;
+    }
+
+    return result;
+}
+
+
```
Comment (tkrokli) [inline src/tools/vmap4_extractor/vmapexport.cpp]: 2 redundant blank lines (3 newlines where 1 is enough).

**Comment by tkrokli** [comment]
Thanks for the cleanup, the codestyle looks better now.

**Comment by tkrokli** [comment]
Just in case you didn't know: the AppVeyor error is not caused by this PR, it is caused by something completely different.

Looks like you removed the last NewLine (LineFeed) in `src/tools/map_extractor/System.cpp` thereby causing the EOF (End of File) marker to be placed on the same line as the last curly bracket: https://github.com/TrinityCore/TrinityCore/pull/20036/files#diff-72dacbe9fe20f6542ab0e87c4f03e6e8R1433
Please add 1 LineFeed / NewLine (or CR/LF in Windows) after the last `}` on  line `src/tools/map_extractor/System.cpp:1433` .

Regarding C++ codestyle in general: https://trinitycore.atlassian.net/wiki/pages/viewpage.action?pageId=2130103#C++DevelopmentStandards-Brackets
> We use symmetric brackets. One line blocks do not need brackets.

Even though you may find examples in the source code where the brackets are not symmetric, it is recommended to use symmetric brackets (if possible).

**Comment by HannibalRoG** [comment]
Can this affect this errors on startup #20075 ?

**Comment by Aokromes** [comment]
No version change on this PR.

File: src/tools/map_extractor/System.cpp [hunk 4]
```diff
@@ -1292,6 +1292,26 @@ void ExtractGameTables()
     printf("Extracted %u files\n\n", count);
 }
 
+
+int GetLocale() {
```
Comment (joschiwald) [inline src/tools/map_extractor/System.cpp]: { should go on a new line

Comment (joschiwald) [inline src/tools/map_extractor/System.cpp]: and move this function to extractor_common project

Comment (tkrokli) [inline src/tools/map_extractor/System.cpp]: Yes, that is true for the rest of the PR too
(general TC recommend C++ codestyle).

**Comment by Shauren** [comment]
Will be fixed slightly differently (requires change in CascLib)

**Comment by Aokromes** [comment]
Properly fixed on 7eab6db

**Pull request closed without merging**